### DISCUSSION TITLE
How to work with DuckDB file from multiple apps

### DISCUSSION THREAD

**Discussion by wrb2** [duckdb/duckdb#8121] [Q&A]
Let's say I have some data I want to transform, and the transformation is complex and of many steps, so I want to use dbt. Normally I'd have DBeaver open (for querying interactively), run dbt from a terminal and have a text editor to edit the dbt code.

With other databases that works. With DuckDB I can't do that, because I can have one database open only from one instance. So if I want to run dbt, I need to close DBeaver (_disconnect_ isn't good enough for some reason, the file stays open).

How are you solving this? It would be even worse if I used Jupyter or something.

**Comment by Alex-Monahan** [comment]
Howdy! One option may be to use another GUI besides DBeaver that may not require a full close and reopen. Another option is to check out Buena Vista! It wraps DuckDB in a Postgres protocol server so you can have multi-process access. https://github.com/jwills/buenavista

One other thought! If you run dbt from the same Python process as your analysis, multi-threaded access works, just not multi process.

**Comment by bill-ash** [comment]
Similar workflow. Running dbt-duckdb from wsl2 and connecting via dbeaver on the windows side. I've been fine with: 

- dbt run 
- Connect to file and check result using dbeaver 
- disconnect 
- edit dbt model 
- dbt run 
- reconnect and check result 
- repeat 

If I forget to disconnect and execute dbt run and try to read from the database, I get a comparability error. Deleting the duckdb file and re-running dbt fixes the issue. 

It's a little bit annoying having to disconnect but honestly it's such a great development pattern I don't mind it. 

Interested in trying the Postgres wrapper. Didn't understand the purpose at first but now I get it!

**Comment by bladko-tom** [comment]
Any chance to allow read-only mode when opening DuckDB DB in DBeaver ? Maybe this would help with this issue.


![image](https://github.com/duckdb/duckdb/assets/112213846/1803e6db-0965-4ae4-8afa-70fbb09e832d)

This caused error 
![image](https://github.com/duckdb/duckdb/assets/112213846/c4981205-0909-421b-868c-0b625cdd05bb)

This is something related to JDBC driver or DBeaver itself?

**Reply by geoHeil** [reply]
Jetbrains can do readonly - must be the driver config from DBeaver

**Reply by starascendin** [reply]
you need to set properties in the driver tab and set it to true

but even that doesn't solve the pooling conncetion issue

**Comment by alanwilter** [comment]
This is how to setup read-only for dbeaver with duckdb https://stackoverflow.com/questions/74411716/set-read-only-connection-to-duckdb-in-dbeaver
